x/ccv/consumer/types/genesis.go

```diff
@@ -109,6 +110,19 @@ func (gs GenesisState) Validate() error {
 		if gs.ProviderClientId == "" {
 			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, "provider client id must be set for a restarting consumer genesis state")
 		}
+		// handshake is still in progress
+		if gs.ProviderChannelId == "" && (len(gs.MaturingPackets) != 0 || len(gs.OutstandingDowntimeSlashing) != 0 || gs.LastTransmissionBlockHeight.Height != 0) {
```

Comment:
Nit: could we extract a variable `handshakeInProgress := ...` and do `if handshakeInProgress {` . It might be more readable.